type-generator/src/main/java/io/mesosphere/types/gen/Main.java

```diff
@@ -0,0 +1,154 @@
+package io.mesosphere.types.gen;
+
+import io.mesosphere.types.gen.compiler.impl.StringTemplateEngine;
+import io.mesosphere.types.gen.compiler.internal.CompiledFragment;
+import io.mesosphere.types.gen.compiler.internal.CompilerEngine;
+import io.mesosphere.types.gen.compiler.internal.CompiledFragmentType;
+import io.mesosphere.types.gen.compiler.internal.arguments.FileObjectTypeFragmentArguments;
+import io.mesosphere.types.gen.models.internal.Type;
+import io.mesosphere.types.gen.models.internal.TypeScope;
+import io.mesosphere.types.gen.models.internal.scopes.FileScope;
+import io.mesosphere.types.gen.models.internal.scopes.GlobalScope;
+import io.mesosphere.types.gen.models.internal.types.*;
+import io.mesosphere.types.gen.models.parser.RAMLParser;
+import io.mesosphere.types.gen.repository.internal.Repository;
+import io.mesosphere.types.gen.repository.internal.RepositoryView;
+import org.apache.commons.cli.*;
+
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.PrintStream;
+import java.util.Map;
+
+public class Main {
+
+    /**
+     * Entry point
+     * @param args
+     */
+    public static void main(String[] args) throws ParseException {
+        Options options = new Options();
+        CommandLineParser parser = new DefaultParser();
+
+        // Prepare options
+        Option o_help = Option.builder("h")
```

Comment:
suggestion: `JCommander` gives an arguably nicer parameter definition (with annotations) and parsing: http://jcommander.org/